opm/simulators/wells/WellGroupHelpers.cpp

```diff
@@ -1181,6 +1186,19 @@ namespace WellGroupHelpers
             }
         }
 
+        // check whether guide rate is violated
+        if(local_reduction_level > 0) {
```

Comment:
Nitpicks:
Space after if.

Below, I think adding an alias
```
const auto& guided_group = chain[local_reduction_level];
```
or similar would improve readability. All variables here can also be const.

(Same for injector function)

Reply:
Thanks for the suggestions.